examples/udpmulticat.c

```diff
@@ -113,158 +206,184 @@ static int catch(struct uprobe *uprobe, struct upipe *upipe,
+                    upipe_err(upipe, "Couldn't read remote addr");
+                    abort();
+                }
+                uint16_t port = ntohs(in.sin_port);
+                char ip[16];
+                inet_ntop(in.sin_family, &in.sin_addr, ip, sizeof(ip));
+                upipe_warn_va(upipe, "%s:%hu", ip, port);
+                char uri[200];
+                char *at = strchr(srcpath, '@');
+                *at++ = '\0';
+                char *sport = strchr(at, ':');
+                *sport++ = '\0';
+
+                snprintf(uri, sizeof(uri), "%s:%hu/srcaddr=%s/srcport=%s/ttl=64", ip, port, at, sport);
+
+                if (!ubase_check(upipe_set_uri(rtcp_sink, uri))) {
+                    upipe_err_va(upipe, "Could not set uri to %s", uri);
+                    upipe_release(upipe_rtpfb_sub);
+                    upipe_release(rtcp_sink);
+                    upipe_rtpfb_sub = NULL;
+                    exit(1);
+                }
+#endif
+            }
+
+            break;
+        }
     }
     return UBASE_ERR_NONE;
 }
 
 int main(int argc, char *argv[])
 {
-    const char *srcpath, *dirpath, *suffix = NULL;
-    bool udp = false;
-    uint64_t rotate = 0;
-    uint64_t rotate_offset = 0;
+    const char *dirpath, *latency, *suffix = NULL;
     int opt;
-    enum uprobe_log_level loglevel = UPROBE_LOG_LEVEL;
 
     /* parse options */
-    while ((opt = getopt(argc, argv, "r:O:d")) != -1) {
+    while ((opt = getopt(argc, argv, "qd")) != -1) {
         switch (opt) {
-            case 'r':
-                rotate = strtoull(optarg, NULL, 0);
-                break;
-            case 'O':
-                rotate_offset = strtoull(optarg, NULL, 0);
-                break;
             case 'd':
-                loglevel = UPROBE_LOG_DEBUG;
+                loglevel--;
+                break;
+            case 'q':
+                loglevel++;
                 break;
             default:
                 usage(argv[0]);
         }
     }
-    if (argc - optind < 2) {
+    if (argc - optind < 3) {
         usage(argv[0]);
     }
     srcpath = argv[optind++];
     dirpath = argv[optind++];
-    if (argc - optind >= 1) {
-        suffix = argv[optind++];
-    } else {
-        udp = true;
-    }
+    latency = argv[optind++];
 
     /* setup environnement */
+
+    struct ev_loop *loop = ev_default_loop(0);
     struct umem_mgr *umem_mgr = umem_alloc_mgr_alloc();
     struct udict_mgr *udict_mgr = udict_inline_mgr_alloc(UDICT_POOL_DEPTH,
                                                          umem_mgr, -1, -1);
     struct uref_mgr *uref_mgr = uref_std_mgr_alloc(UREF_POOL_DEPTH, udict_mgr,
                                                    0);
-    struct upump_mgr *upump_mgr = upump_ev_mgr_alloc_default(UPUMP_POOL,
-            UPUMP_BLOCKER_POOL);
-    struct uclock *uclock = uclock_std_alloc(UCLOCK_FLAG_REALTIME);
+    struct upump_mgr *upump_mgr = upump_ev_mgr_alloc(loop, UPUMP_POOL,
```

Comment:
upump_ev_mgr_alloc_default avoids having to create an ev_loop.

Reply:
I didn't update after the upump changes